src/OrchardCore.Modules/OrchardCore.Setup/Views/_Layout.cshtml

```diff
@@ -21,5 +57,22 @@
             @await RenderSectionAsync("Footer", required: false)
         </footer>
     </div>
+    <script>
+        function setLocalizationCookie() {
+            var culturesList = document.getElementById('culturesList');
+            var culture = culturesList.options[culturesList.selectedIndex].value
+            var url = window.location.origin;
+
+            if (url.indexOf('?') > -1) {
+                url = url.split('&culture=')[0] + '&';
+            }
+            else {
+                url += '?';
+            }
+
+            url += 'culture=' + culture + '&ui-culture=' + culture;
```

Comment:
Not sure if this is required that we set CurrentCulture and CurrentUICulture when we do setup things. I think actually that we should only use the CurrentUICulture for this.

Reply:
No problem `ui-culture` is enought